.github/workflows/doc-linkcheck.yml

```
@@ -0,0 +1,50 @@
+name: doc-linkcheck
+run-name: Documentation Link Check
+
+on:
+  pull_request:
+    types: [labeled]
+
+permissions:
+  contents: read
+
+env:
+  HATCH_ENV_TYPE_VIRTUAL_PATH: ${{ github.workspace }}/.hatch_envs
+
+jobs:
+  linkcheck-docs:
+    name: Sphinx Documentation Link Check
+    runs-on: ubuntu-latest
+    # Run only when the specific label is added
+    if: github.event.label.name == 'status | ready for merge'
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - uses: ./.github/actions/setup-docs-env
+
+      - name: Check GitHub API Rate Limit
+        id: rate_limit_check
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          API_URL="https://api.github.com/rate_limit"
+          RESPONSE=$(curl -s -H "Authorization: token $GH_TOKEN" -H "Accept: application/vnd.github.v3+json" $API_URL)
+          REMAINING_CORE=$(echo "$RESPONSE" | jq -r '.resources.core.remaining')
+          echo "Remaining core API calls: $REMAINING_CORE"
+          THRESHOLD=100
+          if [ "$REMAINING_CORE" -lt "$THRESHOLD" ]; then
+            echo "skip_linkcheck=true" >> $GITHUB_OUTPUT
+            echo "Low GitHub API core rate limit ($REMAINING_CORE remaining). Skipping linkcheck."
```

IIUIC, in this case, the test will pass, it will not run the linkcheck, and there will be a note in its log, correct? I'm afraid that sounds like a silence, we will have no reason to read the log when the test is green, and we will not learn about broken links until we accidentally hit one in live docs. I, for one, would prefer a failed test humans can decide to waive and ignore rather than a machine doing so :/

---

If we consume the 5000 limit and onwhere it fails there is something else wrong. I'm not changing it, but feel free to do so

---

I'm sorry, but I don't understand, is that a yes or no on my assumption that, once reaching the limit, linkcheck would be silently skipped?

---

isn't there to possibility to mark the check as skipped rather the passed in this case?

---

> isn't there to possibility to mark the check as skipped rather the passed in this case?
Nope, it is not possible